### PR TITLE
[ShadowContainer] Cache issue on CornerRadius 

### LINKED ISSUE #782: [ShadowContainer] Cache issue on CornerRadius
## What would you like to be added:

When running the tests for the ShadowsContainer.
We use the a 
Border (Width="200", Height="200"),
image

Rectangle (Width="200", Height="200") and 
image

Border (Width="200", Height="200", CornerRadius="0,200,0,200"),
image

When the test is performed with the corner radius parameters in the border. Shadows assumes the element's corner radius.
However, when we go back to the border element without the corner radius, it is being cut to the shape of the one with the corner radius.

This is causing the test that has the same parameters to fail during execution.

When using 9 and -9.
image

When using 10 and -10
image

The situation only happens when we run the tests at once. Individually tests pass.

## Why is this needed:

It could be a problem when creating the test, but it could be a problem with the cornerRadius cache.

## For which Platform:

- [x] WebAssembly
- [ ] Android
- [ ] iOS
- [ ] macOS (AppKit)
- [ ] Mac Catalyst
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Linux Framebuffer
  - [ ] Tizen
- [ ] Windows

### PR DESCRIPTION
**Pull request by rafael-rosa-knowcode**
GitHub Issue (If applicable): closes #782

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

We do not have cache control for CornerRadius.

## What is the new behavior?

Now we have the cache control for corner radius.

## PR Checklist

Please check if your PR fulfills the following requirements:
- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Tested the changes where applicable:
	- [ ] UWP
	- [ ] WinUI
	- [ ] iOS
	- [ ] Android
	- [ ] WASM
	- [ ] MacOS
- [ ] Updated the documentation as needed:
	- [ ] [General Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc)
	- [ ] [Controls Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc/controls)
	- [ ] [Extensions Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc/helpers)
	- [ ] [controls-styles.md](https://github.com/unoplatform/uno.toolkit.ui/blob/main/doc/controls-styles.md)
	- [ ] [lightweight-styling.md (LightWeight Styling Resource Keys)](https://github.com/unoplatform/uno.toolkit.ui/blob/main/doc/lightweight-styling.md)
- [ ] [Runtime Tests and/or UI Tests](https://platform.uno/docs/articles/contributing/guidelines/creating-tests.html) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal)
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Review by agneszitte — COMMENTED** [review]
Same here can some tests could be added for this ?

File: src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.Paint.cs [hunk 1]
```diff
@@ -72,7 +72,7 @@ private void OnSurfacePainted(object? sender, SKPaintSurfaceEventArgs e)
 		using var _ = canvas.SnapshotState();
 
 		var key =
-			FormattableString.Invariant($"[{contentAsFE.ActualWidth}x{contentAsFE.ActualHeight},{_shadowHost.ActualWidth}x{_shadowHost.ActualHeight},{background}]: ") +
+			FormattableString.Invariant($"[{contentAsFE.ActualWidth}x{contentAsFE.ActualHeight},{_shadowHost.ActualWidth}x{_shadowHost.ActualHeight},{background},{shape}]: ") +
```
Comment (Xiaoy312) [inline src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.Paint.cs:75]: can you add ToString method to various IShadowShapeContext's concrete classes?
to make them more concise than: 
> CornerRadiusRectShadowShapeContext { Width = 5, Height = 10, CornerRadius = 2,2,2,2 }

**Review by Xiaoy312 — APPROVED** [review]

File: src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.Paint.cs (hunk 1 above)
Comment (rafael-rosa-knowcode) [reply src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.Paint.cs]: Done. Thanks.

**Review by agneszitte — APPROVED** [review]

**Comment by agneszitte** [comment]
@Mergifyio backport legacy/3x release/stable/4.2

**Pull request merged**